lib/BackbeatConsumer.js

```diff
@@ -115,14 +115,31 @@ class BackbeatConsumer extends EventEmitter {
                         const bytes = locations.reduce((sum, item) =>
                             sum + item.size, 0);
 
-                        if (!this._metricsStore[qEntry.bucket]) {
-                            this._metricsStore[qEntry.bucket] = {
+                        const repInfo = qEntry.getReplicationInfo();
```

Comment:
This event handler is already quite long and with lots of indentation levels. Can you try putting the metrics code into a separate method? It should help clarity.